src/libcrun/linux.c

```diff
@@ -5667,6 +5667,22 @@ libcrun_kill_linux (libcrun_container_status_t *status, int signal, libcrun_erro
         return libcrun_kill_linux_no_pidfd (status, true, signal, err);
       if (errno == ESRCH)
         return crun_make_error (err, errno, "container not running");
+      /* Check if the PID is valid before reporting an error. */
+      if (errno == EINVAL)
+        {
+          ret = libcrun_check_pid_valid (status, err);
+          if (UNLIKELY (ret < 0))
+            return ret;
+
+          if (ret == 0)
+            {
+              errno = ESRCH;
+              return crun_make_error (err, errno, "kill container");
+            }
+
+          /* Return a specific error message for EINVAL. */
+          return crun_make_error (err, errno, "unsupported flags or invalid pid");
```

Comment:
I like more the default action below, can we drop these two lines here?

You must make sure the `errno` value is preserved and refers to the original one, before calling `libcrun_check_pid_valid`.

You need something like `int errno_saved = errno;` before calling `libcrun_check_pid_valid` and then set it back afterwards.